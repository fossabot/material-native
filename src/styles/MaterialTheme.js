'use strict';
import * as colors from './colors';
import shades from './shades';
const F = Object.freeze;

function normalizeTheme(theme) {
	return theme === 'dark' ? 'dark' : 'light';
}

function inverse(theme) {
	return theme === 'dark' ? 'light' : 'dark';
}

const themeTones = F({
	light: F({
		statusBar: colors.grey300,
		barStyle: 'dark-content',
		appBar: colors.grey100,
		background: colors.grey50,
		container: colors.white,
	}),
	dark: F({
		statusBar: colors.black,
		barStyle: 'light-content',
		appBar: colors.grey900,
		background: '#303030',
		container: colors.grey800,
	}),
});

export default class MaterialTheme {
	static defaultTheme = new MaterialTheme();

	constructor(settings) {
		this.settings = settings || {};
		// @todo recursively freeze settings
		this._applyDefaults();
		this._buildTheme();
		this._freeze();
	}

	_applyDefaults() {
		this.settings.theme = normalizeTheme(this.settings.theme);
	}

	_buildTheme() {
		const {theme} = this.settings;
		const inverseTheme = inverse(theme);

		this.themeTone = theme;
		this.inverseThemeTone = inverseTheme;

		this.palette = {
			primary: this.settings.primary || colors.indigo500,
			darkPrimary: this.settings.darkPrimary || colors.indigo700,
			accent: this.settings.accent || colors.pinkA200,
			background: themeTones[theme].background,
			container: themeTones[theme].container,
		};

		this.defaultToolbar = Object.assign({
			theme,
			background: themeTones[theme].appBar,
		}, this.settings.defaultToolbar || {});

		// @todo Automatically pick the right appBar theme based on the appBar background
		//       Many solid appBar colors should implicitly turn the theme within the appBar to dark
		const appBarTheme = normalizeTheme((this.settings.appBar && this.settings.appBar.theme) || this.defaultToolbar.theme);
		this.appBar = Object.assign({}, this.defaultToolbar, {
			theme: appBarTheme,
			background: this.palette.primary || this.defaultToolbar.background,
		}, this.settings.appBar || {});

		this.statusBar = Object.assign({
			background: this.palette.darkPrimary || themeTones[inverse(appBarTheme)].statusBar,
			translucentBackground: shades.dark.statusBar,
			barStyle: themeTones[inverse(appBarTheme)].barStyle,
		}, this.settings.statusBar || {});

		this.text = Object.assign({
			primaryColor: shades[inverseTheme].primaryText,
			secondaryColor: shades[inverseTheme].secondaryText,
			disabledColor: shades[inverseTheme].disabledText,
			hintColor: shades[inverseTheme].hintText,
			primaryOpacity: shades.opacity[inverseTheme].primaryText,
			secondaryOpacity: shades.opacity[inverseTheme].secondaryText,
			disabledOpacity: shades.opacity[inverseTheme].disabledText,
			hintOpacity: shades.opacity[inverseTheme].hintText,
		}, this.settings.text || {});

		this.icon = Object.assign({
			activeColor: shades[inverseTheme].activeIcon,
			inactiveColor: shades[inverseTheme].inactiveIcon,
			activeOpacity: shades.opacity[inverseTheme].activeIcon,
			inactiveOpacity: shades.opacity[inverseTheme].inactiveIcon,
		}, this.settings.icon || {});
	}

	_freeze() {
		Object.freeze(this); // @todo recursive
	}

	/**
	 * Create a new theme object that inherits settings from another
	 */
	extend(settings) {
		// @todo immutably merge `this.settings` with `settings` and pass that to `new MaterialTheme()`
	}
}

export function getMaterialTheme(settings) {
	return new MaterialTheme(settings);
}
