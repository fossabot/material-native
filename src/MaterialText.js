'use strict';
import React, {PureComponent} from 'react';
import {StyleSheet, Text} from 'react-native';
import withMaterialTheme from './styles/withMaterialTheme';
import * as typo from './styles/typo';
import shades from './styles/shades';

/**
 * Horizontal MaterialText line
 */
class MaterialText extends PureComponent {
	render() {
		const {
			materialTheme,
			style,
			// Typo class
			display4,
			display3,
			display2,
			display1,
			headline,
			title,
			subhead,
			body2,
			body1, // eslint-disable-line no-unused-vars
			caption,
			button,
			// Script size
			dense,
			tall,
			// Shade / color
			secondary,
			disabled,
			hint,
			color: colorOverride,
			// Theme
			dark,
			light,
			...textProps, // eslint-disable-line comma-dangle
		} = this.props;

		let typoClass, typoClasses, color;

		if ( dense ) typoClasses = typo.dense;
		else if ( tall ) typoClasses = typo.tall;
		else typoClasses = typo;

		if ( display4 ) typoClass = 'display4';
		else if ( display3 ) typoClass = 'display3';
		else if ( display2 ) typoClass = 'display2';
		else if ( display1 ) typoClass = 'display1';
		else if ( headline ) typoClass = 'headline';
		else if ( title ) typoClass = 'title';
		else if ( subhead ) typoClass = 'subhead';
		else if ( body2 ) typoClass = 'body2';
		else if ( caption ) typoClass = 'caption';
		else if ( button ) typoClass = 'button';
		else typoClass = 'body1';

		if ( colorOverride ) {
			color = colorOverride;
		} else if ( dark || light ) {
			const theme = dark ? 'dark' : 'light';
			if ( secondary ) color = shades[theme].secondaryText;
			else if ( disabled ) color = shades[theme].disabledText;
			else if ( hint ) color = shades[theme].hintText;
			else color = shades[theme].primaryText;
		} else {
			if ( secondary ) color = materialTheme.text.secondaryColor;
			else if ( disabled ) color = materialTheme.text.disabledColor;
			else if ( hint ) color = materialTheme.text.hintColor;
			else color = materialTheme.text.primaryColor;
		}

		return (
			<Text
				{...textProps}
				style={[
					{
						...typoClasses[typoClass],
						color,
					},
					style
				]} />
		);
	}
}

export default withMaterialTheme(MaterialText);
