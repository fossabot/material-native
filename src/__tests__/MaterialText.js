'use strict';
import React from 'react';
import MaterialTheme from '../styles/MaterialTheme';
import ThemeProvider from '../styles/ThemeProvider';
import MaterialText from '../MaterialText';

import ReactTestRenderer from 'react-test-renderer';
import ToJSX from 'react-element-to-jsx-string';

const theme = new MaterialTheme({});

/* eslint-disable react/jsx-key */
function typoClassSnapshots(props) {
	const typoClasses = [
		'display4',
		'display3',
		'display2',
		'display1',
		'headline',
		'title',
		'subhead',
		'body2',
		'body1',
		'caption',
		'button'
	];

	return typoClasses.map((typoClass) => (
		<MaterialText {...props} {...{[typoClass]: true}}>...</MaterialText>
	));
}

const snapshotTests = [
	<MaterialText>...</MaterialText>,
	<MaterialText secondary>...</MaterialText>,
	<MaterialText disabled>...</MaterialText>,
	<MaterialText hint>...</MaterialText>,
	<MaterialText dark>...</MaterialText>,
	<MaterialText dark secondary>...</MaterialText>,
	<MaterialText dark disabled>...</MaterialText>,
	<MaterialText dark hint>...</MaterialText>,
	<MaterialText light>...</MaterialText>,
	<MaterialText light secondary>...</MaterialText>,
	<MaterialText light disabled>...</MaterialText>,
	<MaterialText light hint>...</MaterialText>,
	<MaterialText color='#123456'>...</MaterialText>,
	...typoClassSnapshots({}),
	...typoClassSnapshots({dense: true}),
	...typoClassSnapshots({tall: true})
];
/* eslint-enable react/jsx-key */

for ( const root of snapshotTests ) {
	test(ToJSX(root, {maxInlineAttributesLineLength: Infinity}), () => {
		const host = ReactTestRenderer.create(
			<ThemeProvider theme={theme}>
				{root}
			</ThemeProvider>
		);
		expect(host.toJSON()).toMatchSnapshot();
	});
}
