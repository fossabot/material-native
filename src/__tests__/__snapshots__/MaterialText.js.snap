// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withMaterialTheme(MaterialText) body1 dense>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) body1 tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) body1>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) body2 dense>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "500",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) body2 tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) body2>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "500",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) button dense>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "500",
        "letterSpacing": 0.018,
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) button tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "600",
        "letterSpacing": 0.018,
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) button>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.018,
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) caption dense>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 13,
        "fontWeight": "400",
        "letterSpacing": 0.011,
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) caption tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 13,
        "fontWeight": "400",
        "letterSpacing": 0.011,
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) caption>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0.011,
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) color="#123456">
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#123456",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dark disabled>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.38)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dark hint>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.38)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dark secondary>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.54)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dark>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dense display1>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 34,
        "fontWeight": "400",
        "letterSpacing": -0.01,
        "lineHeight": 40,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dense display2>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 45,
        "fontWeight": "400",
        "letterSpacing": -0.018,
        "lineHeight": 48,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dense display3>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 56,
        "fontWeight": "400",
        "letterSpacing": -0.026,
        "lineHeight": 60,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dense display4>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 112,
        "fontWeight": "300",
        "letterSpacing": -0.044,
        "lineHeight": 120,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dense headline>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 24,
        "fontWeight": "400",
        "letterSpacing": -0.012,
        "lineHeight": 32,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dense subhead>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) dense title>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 21,
        "fontWeight": "500",
        "lineHeight": 28,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) disabled light>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(255, 255, 255, 0.5)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) disabled>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.38)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) display1 tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 34,
        "fontWeight": "400",
        "letterSpacing": -0.01,
        "lineHeight": 40,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) display1>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 34,
        "fontWeight": "400",
        "letterSpacing": -0.01,
        "lineHeight": 40,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) display2 tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 45,
        "fontWeight": "400",
        "letterSpacing": -0.018,
        "lineHeight": 48,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) display2>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 45,
        "fontWeight": "400",
        "letterSpacing": -0.018,
        "lineHeight": 48,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) display3 tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 56,
        "fontWeight": "400",
        "letterSpacing": -0.026,
        "lineHeight": 60,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) display3>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 56,
        "fontWeight": "400",
        "letterSpacing": -0.026,
        "lineHeight": 60,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) display4 tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 112,
        "fontWeight": "400",
        "letterSpacing": -0.044,
        "lineHeight": 120,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) display4>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 112,
        "fontWeight": "300",
        "letterSpacing": -0.044,
        "lineHeight": 120,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) headline tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 24,
        "fontWeight": "400",
        "letterSpacing": -0.012,
        "lineHeight": 32,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) headline>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 24,
        "fontWeight": "400",
        "letterSpacing": -0.012,
        "lineHeight": 32,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) hint light>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(255, 255, 255, 0.5)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) hint>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.38)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) light secondary>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(255, 255, 255, 0.7)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) light>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) secondary>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.54)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) subhead tall>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) subhead>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) tall title>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 21,
        "fontWeight": "600",
        "lineHeight": 28,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText) title>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "500",
        "lineHeight": 28,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;

exports[`<withMaterialTheme(MaterialText)>
  ...
</withMaterialTheme(MaterialText)> 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  ...
</Text>
`;
